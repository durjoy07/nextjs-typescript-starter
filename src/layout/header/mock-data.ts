import { IHeader } from './interface';

export const headerProps: IHeader = {
  logo: 'Lemon Hive',
  menuItems: [
    {
      label: 'Home',
      href: '/',
    },
    {
      label: 'About',
      href: '/',
    },
    {
      label: 'Service',
      href: '/',
    },
    {
      label: 'Blog',
      href: '/',
    },
  ],
};
