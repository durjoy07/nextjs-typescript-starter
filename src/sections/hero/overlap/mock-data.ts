import { IHeroWithOverlapContent } from './interface';

export const heroWithOverlapContentProps: IHeroWithOverlapContent = {
  title: 'Lemon Hive, We Build Lightning Fast Websites',
  description:
    'We Build Lightning Fast Websites With Amazing UX Using Headless Architectures',
  buttons: {
    buttonLeft: {
      label: 'Server Side Example',
      href: '/',
    },
    buttonRight: {
      label: 'Client Side Example',
      href: 'client-side',
    },
  },
};
