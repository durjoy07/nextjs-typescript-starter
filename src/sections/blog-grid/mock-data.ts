import { IBlogGrid } from './interface';
import { blogCardProps } from '@components/cards/blog/mock-data';
import { sectionDescriptionProps } from '@components/section-heading/mock-data';

export const blogGridProps: IBlogGrid = {
  sectionHeading: {
    title: 'Our Blog',
    description: sectionDescriptionProps,
  },
  blogs: [
    { ...blogCardProps, tags: ['Nextjs', 'Reactjs'] },
    { ...blogCardProps, tags: ['Nextjs', 'Reactjs'] },
    { ...blogCardProps, tags: ['Nextjs'] },
    { ...blogCardProps },
    { ...blogCardProps },
    { ...blogCardProps },
    { ...blogCardProps },
    { ...blogCardProps },
  ],
};
