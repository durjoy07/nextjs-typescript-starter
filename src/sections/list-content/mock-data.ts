import { sectionDescriptionProps } from '@components/section-heading/mock-data';
import { ISanityCustomPortableText } from '@root/src/backend/types/portable-text';
import { IListContentSection } from './interface';

export const descriptionPortableData: ISanityCustomPortableText = [
  {
    _key: 'd956675f69e0',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'For design, branding, web development, marketing, consulting & other agencies.',
        _key: '612b8d7ad65a0',
      },
    ],
    markDefs: [],
    _type: 'customBlock',
    style: 'normal',
  },
  {
    _key: 'f9a72086f01c',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'We work with agencies in a number of different ways, but predominantly as an extension of your team to add headless development expertise.',
        _key: '3ec390abb3560',
      },
    ],
    markDefs: [],
    _type: 'customBlock',
    style: 'normal',
  },
  {
    _key: 'af14faef2a19',
    children: [
      {
        _type: 'span',
        marks: [],
        text: 'With the move towards headless website development, it can be difficult to find the right people, with the right expertise, to produce websites to your standards. By outsourcing the development and/or design to us, we can ensure you build exceptional headless websites that exceed what is possible with monolithic platforms such as WordPress.',
        _key: '57de12dc89fc0',
      },
    ],
    markDefs: [],
    _type: 'customBlock',
    style: 'normal',
  },
];

export const listContentSectionProps: IListContentSection = {
  sectionTheme: 'secondary',
  sectionHeading: {
    title: 'Who We Work With',
    description: sectionDescriptionProps,
  },
  lists: [
    {
      title: 'Services for Agencies',
      description: descriptionPortableData,
    },
    {
      title: 'Services for Agencies',
      description: descriptionPortableData,
    },
  ],
};
