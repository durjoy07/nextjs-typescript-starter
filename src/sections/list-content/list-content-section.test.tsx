import { render } from '@testing-library/react';
import ListContentSection from '.';
import { listContentSectionProps } from './mock-data';

it('renders list content section correctly', () => {
  const { getAllByRole } = render(
    <ListContentSection {...listContentSectionProps} />
  );
  const element = getAllByRole('presentation')[0];

  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
