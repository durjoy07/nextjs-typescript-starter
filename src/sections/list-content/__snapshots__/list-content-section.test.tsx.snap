// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list content section correctly 1`] = `
<section
  class="section-padding-primary secondary"
  role="presentation"
>
  <div
    class="container"
    role="presentation"
  >
    <div
      class="w-full max-w-[800px]"
    >
      <div
        class="space-y-2 lg:space-y-3 mb-[1.875rem] md:mb-[3.25rem] section-heading secondary"
        role="presentation"
      >
        <h2
          class="text-[1.75rem] font-medium leading-[1.3] md:text-[3rem] md:font-bold md:leading-[1.1] title"
        >
          Who We Work With
        </h2>
        <div
          class="text-[.875rem] leading-[1.3] md:text-base mt-3 lg:mt-5 description"
        >
          <article
            class="prose max-w-full space-y-4"
          >
            <p
              class="dark:text-white"
              id="0ffdc2e0f0b8"
            >
              <a
                class="no-underline styled-link"
                href="https://www.google.com"
                target="_blank"
              >
                <span
                  class="link-underline"
                >
                  Headless website
                </span>
              </a>
               for an award winning digital marketing agency.
            </p>
            <br />
            <p
              class="dark:text-white"
              id="c0b3d76ed804"
            >
              <strong>
                Lorem ipsum
              </strong>
               dolor 
              <em>
                sit amet consectetur
              </em>
              . Vitae cursus ultricies interdu ridiculus. Nullam nuwel craegestas risus 
              <span
                style="text-decoration: underline;"
              >
                diam lorem nascetur
              </span>
              . Lorem ipsum dolor sit amet consectetur. Vitae cursus ultricies interdu ridiculus. Nullam nuwel craegestas risus 
              <strong>
                diam lorem nascetu ullam nuwel
              </strong>
            </p>
          </article>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 gap-10 lg:grid-cols-2 lg:gap-12"
    >
      <div
        class="space-y-4"
      >
        <h3
          class="text-lg lg:text-2xl font-bold theme-text"
        >
          Services for Agencies
        </h3>
        <div>
          <article
            class="prose max-w-full space-y-4"
          >
            <p
              class="dark:text-white"
              id="d956675f69e0"
            >
              For design, branding, web development, marketing, consulting & other agencies.
            </p>
            <p
              class="dark:text-white"
              id="f9a72086f01c"
            >
              We work with agencies in a number of different ways, but predominantly as an extension of your team to add headless development expertise.
            </p>
            <p
              class="dark:text-white"
              id="af14faef2a19"
            >
              With the move towards headless website development, it can be difficult to find the right people, with the right expertise, to produce websites to your standards. By outsourcing the development and/or design to us, we can ensure you build exceptional headless websites that exceed what is possible with monolithic platforms such as WordPress.
            </p>
          </article>
        </div>
      </div>
      <div
        class="space-y-4"
      >
        <h3
          class="text-lg lg:text-2xl font-bold theme-text"
        >
          Services for Agencies
        </h3>
        <div>
          <article
            class="prose max-w-full space-y-4"
          >
            <p
              class="dark:text-white"
              id="d956675f69e0"
            >
              For design, branding, web development, marketing, consulting & other agencies.
            </p>
            <p
              class="dark:text-white"
              id="f9a72086f01c"
            >
              We work with agencies in a number of different ways, but predominantly as an extension of your team to add headless development expertise.
            </p>
            <p
              class="dark:text-white"
              id="af14faef2a19"
            >
              With the move towards headless website development, it can be difficult to find the right people, with the right expertise, to produce websites to your standards. By outsourcing the development and/or design to us, we can ensure you build exceptional headless websites that exceed what is possible with monolithic platforms such as WordPress.
            </p>
          </article>
        </div>
      </div>
    </div>
  </div>
</section>
`;
